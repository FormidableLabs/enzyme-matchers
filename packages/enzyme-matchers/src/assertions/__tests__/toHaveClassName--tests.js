const { shallow, mount } = require('enzyme');
const React = require('react');

const toHaveClassName = require('../toHaveClassName');

function Fixture() {
  return (
    <div>
      <span className="foo" />
      <span className="bar baz" />
    </div>
  );
}

describe('toHaveClassName', () => {
  [shallow, mount].forEach(builder => {
    describe(builder.name, () => {
      const wrapper = builder(<Fixture />);
      const truthyResults = toHaveClassName(wrapper.find('.bar'), 'bar');
      const falsyResults = toHaveClassName(wrapper.find('.bar'), 'asldfkj');

      it('returns the pass flag properly', () => {
        expect(truthyResults.pass).toBeTruthy();
        expect(falsyResults.pass).toBeFalsy();
      });

      it(`returns the message with the proper pass verbage (${builder.name})`, () => {
        expect(truthyResults.message).toMatchSnapshot();
      });

      it(`returns the message with the proper fail verbage (${builder.name})`, () => {
        expect(truthyResults.negatedMessage).toMatchSnapshot();
      });

      it(`provides contextual information for the message (${builder.name})`, () => {
        expect(truthyResults.contextualInformation).toMatchSnapshot();
      });
    });
  });
});
