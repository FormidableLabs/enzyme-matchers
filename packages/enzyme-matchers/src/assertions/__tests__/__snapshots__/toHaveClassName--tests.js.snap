exports[`toHaveClassName mount provides contextual information for the message (mount) 1`] = `
Object {
  "actual": "Found node output: <span class=\"bar baz\"></span>",
}
`;

exports[`toHaveClassName mount returns the message with the proper fail verbage (mount) 1`] = `"Expected <span> to have className of \".bar\" but instead found \".bar baz\""`;

exports[`toHaveClassName mount returns the message with the proper pass verbage (mount) 1`] = `"Expected <span> to have className of \".bar\" but instead found \".bar baz\""`;

exports[`toHaveClassName shallow provides contextual information for the message (shallow) 1`] = `
Object {
  "actual": "Found node output: <span class=\"bar baz\"></span>",
}
`;

exports[`toHaveClassName shallow returns the message with the proper fail verbage (shallow) 1`] = `"Expected <span> to have className of \".bar\" but instead found \".bar baz\""`;

exports[`toHaveClassName shallow returns the message with the proper pass verbage (shallow) 1`] = `"Expected <span> to have className of \".bar\" but instead found \".bar baz\""`;
